### ISSUE TITLE
[FEATURE] Payments and it's webhook response to contain non-card payment method details

### ISSUE THREAD

**Issue by kashif-m** [juspay/hyperswitch#5652]
### Feature Description

TLDR; Payments response and it's associated webhooks events to contain payment method data details.

As of today, only card's and paylater's payment method data is populated in our API responses and webhook events for payments. Populating details of other payment methods provides more context about the payment details.

### Possible Implementation

Below is the list of payment methods supported in HyperSwitch

- :white_check_mark: Card
- CardRedirect
- Wallet
- :white_check_mark: PayLater
- BankRedirect
- BankDebit
- BankTransfer
- RealTimePayment
- Crypto
- Upi
- Voucher
- GiftCard
- CardToken
- OpenBanking

### Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

Yes, I am willing to submit a PR!

**Comment by kashif-m** [comment]
Opening for https://github.com/juspay/hyperswitch/pull/5788

**Comment by kashif-m** [comment]
Reopening for https://github.com/juspay/hyperswitch/pull/5790
